Sources/_StringProcessing/Regex/Options.swift

```diff
@@ -12,17 +12,221 @@
+  /// value to any quantification method to change its behavior.
+  ///
+  /// - Parameter useReluctantCaptures: A Boolean value indicating whether
+  ///   quantifiers should be reluctant by default.
+  public func reluctantCaptures(_ useReluctantCaptures: Bool = true) -> Regex<Output> {
+    wrapInOption(.reluctantByDefault, addingIf: useReluctantCaptures)
+  }
+  
+  /// Returns a regular expression where the `\X`, `\y`, and `\Y` metacharacters
+  /// operate at the word boundary level instead of the grapheme cluster level.
+  ///
+  /// This method corresponds to applying the `y{w}` or `y{g}` option in a
+  /// regular expression literal, and only applies to regular expressions
+  /// specified as literals.
+  ///
+  /// - Parameter useWordBoundaries: A Boolean value indicating whether `\X`,
+  ///   `\y`, and `\Y` should use word boundaries instead of grapheme cluster
+  ///   boundaries.
+  public func segmentTextByWords(_ useWordBoundaries: Bool = true) -> Regex<Output> {
+    wrapInOption(.textSegmentWordMode, addingIf: useWordBoundaries)
+  }
+}
+
+// Options that affect parsing, and therefore likely can't be API
+@available(*, unavailable)
+extension RegexComponent {
+  /// Returns a regular expression where whitespace is not significant for
+  /// matching.
+  ///
+  /// This method corresponds to applying the `xx` option in a regular
+  /// expression literal, and only applies to regular expressions specified as
+  /// literals.
+  ///
+  /// - Parameter useExtendedSyntax: A Boolean value indicating whether
+  ///   whitespace in a regex literal should be ignored.
+  public func extendedSyntax(_ useExtendedSyntax: Bool = true) -> Regex<Output> {
+    wrapInOption(.extraExtended, addingIf: useExtendedSyntax)
+  }
+  
+  /// Returns a regular expression where only named groups are captured.
+  ///
+  /// This method corresponds to applying the `n` option in a regular
+  /// expression literal, and only applies to regular expressions specified as
+  /// literals. In the `RegexBuilder` syntax, use the `Capture` method only
+  /// when you want to capture a portion of a match.
+  ///
+  /// - Parameter namedCapturesOnly: A Boolean value indicating whether only
+  ///   named groups should be captured.
+  public func namedCapturesOnly(_ namedCapturesOnly: Bool = true) -> Regex<Output> {
+    wrapInOption(.noAutoCapture, addingIf: namedCapturesOnly)
+  }
+  
+  /// Returns a regular expression where multiple groups are allowed to share
+  /// the same name.
+  ///
+  /// This method corresponds to applying the `J` option in a regular
+  /// expression literal, and only applies to regular expressions specified as
+  /// literals.
+  ///
+  /// - Parameter allowDuplicateNames: A Boolean value indicating whether groups
+  ///   are allowed to share names.
+  public func allowDuplicateNames(_ allowDuplicateNames: Bool = true) -> Regex<Output> {
+    wrapInOption(.allowDuplicateGroupNames, addingIf: allowDuplicateNames)
+  }
+}
```

Comment:
@milseman @hamishknight Am I right that we can't provide these at the API level? Would there be any value in providing them as deprecated or something so that we can explain the unavailability vs the other methods?